Only in a/docs: busybox.1
Only in a/docs: BusyBox.html
Only in a/docs: busybox.net
Only in a/docs: busybox.pod
Only in a/docs: BusyBox.txt
File b/.git is a directory while file a/.git is a regular file
diff -ur b/include/libbb.h a/include/libbb.h
--- b/include/libbb.h	2016-01-29 22:32:47.083066705 -0600
+++ a/include/libbb.h	2016-01-29 20:38:38.107062013 -0600
@@ -1839,7 +1839,7 @@
  * use bb_default_login_shell and following defines.
  * If you change LIBBB_DEFAULT_LOGIN_SHELL,
  * don't forget to change increment constant. */
-#define LIBBB_DEFAULT_LOGIN_SHELL  "-/bin/sh"
+#define LIBBB_DEFAULT_LOGIN_SHELL  "-/system/bin/sh"
 extern const char bb_default_login_shell[] ALIGN1;
 /* "/bin/sh" */
 #define DEFAULT_SHELL              (bb_default_login_shell+1)
diff -ur b/init/init.c a/init/init.c
--- b/init/init.c	2016-01-29 22:32:47.087066720 -0600
+++ a/init/init.c	2016-01-29 20:38:38.107062013 -0600
@@ -1072,7 +1072,7 @@
 	/* Make sure environs is set to something sane */
 	putenv((char *) "HOME=/");
 	putenv((char *) bb_PATH_root_path);
-	putenv((char *) "SHELL=/bin/sh");
+	putenv((char *) "SHELL=/system/bin/sh");
 	putenv((char *) "USER=root"); /* needed? why? */
 
 	if (argv[1])
diff -ur b/shell/ash.c a/shell/ash.c
--- b/shell/ash.c	2016-01-29 22:32:47.119066839 -0600
+++ a/shell/ash.c	2016-01-29 22:23:32.216913271 -0600
@@ -13188,20 +13188,18 @@
 
 	if (argv[0] && argv[0][0] == '-')
 		isloginsh = 1;
+	else
+		isloginsh = 1;
 	if (isloginsh) {
 		const char *hp;
 
 		state = 1;
-		read_profile("/etc/profile");
+		read_profile("/system/etc/profile");
  state1:
 		state = 2;
-		hp = lookupvar("HOME");
-		if (hp) {
-			hp = concat_path_file(hp, ".profile");
-			read_profile(hp);
-			free((char*)hp);
+		if (!hp){
+		setvar("HISTFILE", "/data/ash_history", 0);
 		}
-	}
  state2:
 	state = 3;
 	if (
@@ -13233,12 +13231,7 @@
 		if (iflag) {
 			const char *hp = lookupvar("HISTFILE");
 			if (!hp) {
-				hp = lookupvar("HOME");
-				if (hp) {
-					hp = concat_path_file(hp, ".ash_history");
-					setvar0("HISTFILE", hp);
-					free((char*)hp);
-					hp = lookupvar("HISTFILE");
+				setvar("HISTFILE", "/data/ash_history", 0);
 				}
 			}
 			if (hp)
